src/functional/ffront/ast_passes/single_static_assign.py

```diff
@@ -48,11 +58,17 @@ def foo():
         a__2 = 3 + a__1
         return a__2
 
-    Note that each variable name is assigned only once and never updated / overwritten.
+    Note that each variable name is assigned only once (per branch) and never updated / overwritten.
 
     Note also that after parsing, running the pass and unparsing we get invalid but
     readable python code. This is ok because this pass is not intended for
     python-to-python translation.
+
+    WARNING: This pass is not intended as a general-purpose SSA transformation.
+    The pass does not support any general Python AST. Known limitations include:
+        * Nested functions aren't supported
+        * Returns inside if blocks aren't supported
```

Comment:
Does this pass care about the returns? For later parts of the pipeline this is true, but the SSA property should still hold even for such cases.

Reply:
Well for once, it mayb add dead code like:

```python
a = 0
if True:
   a = 1
else:
   return 2
```
->
```python
a__0 = 0
if True:
   a__1 = 1
else:
   return 2
   a__1 = a__0
```

and secondly, it will break definite assignment analysis:

```python
if True:
   a = 1
else:
   return x
# because the else branch returned, `a` is guaranteed to be defined here
b = a
```
Currently, this is not supported and will become:
```python
if True:
   a__0 = 1
else:
   return x
b__0 = a
```

I might add support for returns, but I first want to continue with the foast parser & lowering.

Reply:
SSA pass supports returns now, changed doc accordingly. Removed the line about returns.